### PR TITLE
Paper: Performing Netlist Analysis and Transformations Using SpyDrNet

### PR DESCRIPTION
**Pull request by thunder-hammer**
If you are creating this PR in order to submit a draft of your paper,
see http://procbuild.scipy.org/ for logs generated by the build
process.

See the [project readme](https://github.com/scipy-conference/scipy_proceedings#build-your-paper)
for more information.

### CODE REVIEW

**Comment by thunder-hammer** [comment]
@andrewmkeller Wanted you to have the opportunity to see this PR.

**Comment by laserkelvin** [comment]
Hi Team!

Sorry for the belated introduction; I'm going to be helping out with the review for _Performing Netlist Analysis and Transformations Using SpyDrNet_. I'll be posting my comments in the next day or two, and we'll go from there :)

**Comment by laserkelvin** [comment]
Overall I think the paper reads quite well, although there are small stylistic changes that I may recommend (for example fragmented sentences) that might help with the flow, but I'll reserve those for if the authors are interested.

## Typographical comments

- On page 2, subsection _Constructs Employed_, there is a broken reference to a Figure. Prior to this reference there is also a missing space:

> ...Instances, Ports,and Cables.

- On page 4, Uniquify is capitalized in one instance. I suggest using a different type when referencing the API, perhaps as title case or a monospaced font to differentiate between usages.

> Currently Uniquify is implemented to ensure...

- On page 4, there is an extra space before the period in this sentence:

> ...behaviour consistent at all levels of cloning .

- The acronym FPGA is re-defined in the `Applications` section (bottom left, page 5). Simultaneously, SRAM is never defined.

- Triple Modular Redundancy should be defined as an acronym like the others in its section, so consider rephrasing:

> Triple modular redundancy (TMR) is one method... 

- Bottom right of page 5, replace the numeral "2" with "two", according to Strunk & White (3rd edition at least)

> Figure 6 shows 2 typical layouts...

- Top left of page 6 has a fragmented sentence that should be corrected:

> Subsections of the circuit have been chosen to be replicated. Then a voter insertion algorithm...

- Extra period inserted after the BYU EDIF Tools citation:

> Previous work with the BYU EDIF Tools [BYU]. required a flattened design...

- Duplication With Compare should be defined as a regular acronym, see my comment about TMR.

---

## General comments

- In the introduction, along the passage of:

> SpyDrNet differs from most related tools in that its focus...

Would it be possible to provide one or two references to software that are commonly used? For example, I would consider other programs being closed-source and commercial a significant downside, compared to open-source `SpyDrNet`. For example, I know the [CASPER collaboration](https://casper-toolflow.readthedocs.io/en/latest/) implements some workflows for designing and programming FPGAs for radio interferometry, albeit frustratingly built on top of `Matlab` (and therefore a big pass for me). Is there any kind of overlap with `SpyDrNet`?

- Would it be possible to include some snippets of `SpyDrNet` in use, just to kind of show off the API perhaps? For example, it would help with both conceptual and use case understanding for the reader to be able to include short excerpts of concrete code to demonstrate the workflow; even if it's just  copied from the repo README

- In each of the subsections, you could envision appending a code snippet to kind of bring all the ideas discussed in the section together in a pragmatic way—here's the abstraction, and here's the code.

- Following the `Applications` section, is it appropriate to add a `Future Directions` section? I would very much be interested in how `SpyDrNet` is planned to evolve over the next few years.

**Comment by rouseguy** [comment]
Hello everyone,

Sincere apologies for the delayed introduction. I will be helping out in the review process. I will share my initial set of comments by tomorrow.

**Comment by deniederhut** [comment]
Hey @thunder-hammer ! Thanks to some awesome behind the scenes work by @stargaser, we now have experimental support for publishing papers with ORCIDs. If you have one and would like to add it to your paper, it's as simple as adding an `:orcid:` tag under your name in the paper header. [You can see an example of how to do this in the `00_vanderwalt` example paper](https://raw.githubusercontent.com/scipy-conference/scipy_proceedings/2020/papers/00_vanderwalt/00_vanderwalt.rst). If you don't have one, no worries! You can still publish your paper as is.

**Comment by deniederhut** [comment]
Hi @rouseguy ! Have you had a chance to finish up your initial review?

**Comment by rouseguy** [comment]
This is a well-written paper. These are some of my review suggestions to improve the readability of the paper

- In related work section: I would start with something more positive:
"SpyDrNet is not the first tool of its kind.". Is SpyDrNet the first Python library? 
- I would want to show/discuss some unique problem statements that SpyDrNet is trying to solve. I can sense that it was non-trivial to do the tasks outlined in the Introduction section - before SpyDrNet. Specifically calling some of these things out will be good for someone reading the paper to understand where SpyDrNet fits in the ecosystem.
- I would love to see examples. Especially in the section where the author talks about utility functions - it would be good to show how those features (API, hierarchy etc) - provide ease, power and flexibility to hardware engineers/end-users in their workflow.

**Comment by deniederhut** [comment]
Hi @thunder-hammer ! Have you finished addressing comments from the reviewers?

**Comment by deniederhut** [comment]
Hi everyone! Friendly reminder that the open review period ends 2 July -- the committee will be making final decisions on each paper this Friday.

**Comment by thunder-hammer** [comment]
Thank you all for the helpful reviews. We are currently working on getting these suggestions addressed. We plan to update the pull request tomorrow.

**Comment by laserkelvin** [comment]
Thanks @thunder-hammer. Once you've pushed the changes I'll give it another pass.

**Comment by thunder-hammer** [comment]
Thank you again for the suggestions. We believe that we addressed all of the recommendations but if there is something we missed or there is something we could improve please let us know.

**Comment by andrewmkeller** [comment]
Hey @laserkelvin and @rouseguy. Thank you so much for your feedback. I believe that the paper is much improved now because of the feedback you both have provided. We were careful to address each point of feedback and I hope we didn't miss anything. As you can see, we included a number of additional changes to improve the quality and flow of the paper. I hope that you find these changes to be acceptable. Thank you for the time you have taken in reviewing this paper.

File: papers/dallin_skouson/mybib.bib [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+  keywords={error detection;field programmable gate arrays;SRAM chips;online error detection;SRAM-based FPGA;single-event upsets;mitigation strategies;error correction methods;duplication with compare;fault injection;radiation testing;Field programmable gate arrays;Single event transient;Fault detection;Circuit faults;Error correction;Electrical fault detection;Radiation detectors;Testing;Vents;Costs},
+  doi={10.1109/AERO.2008.4526470},
+  ISSN={1095-323X},
+  month={March},}
+
+@misc{BYUediftools,
```
Comment (cbcunc) [inline papers/dallin_skouson/mybib.bib]: Please provide a full citation. https://www.mendeley.com/guides/web-citation-guide

File: papers/dallin_skouson/mybib.bib [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+title={{BYU EDIF Tools}},
+note={https://sourceforge.net/projects/byuediftools/}
+}
+
+@misc{vivado,
+title={{Vivado Design Suite}},
+note={https://www.xilinx.com/products/design-tools/vivado.html}
+}
+
+@misc{verific,
```
Comment (cbcunc) [inline papers/dallin_skouson/mybib.bib]: Please provide a full citation. https://www.mendeley.com/guides/web-citation-guide

File: papers/dallin_skouson/mybib.bib [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+  doi={10.1109/AERO.2008.4526470},
+  ISSN={1095-323X},
+  month={March},}
+
+@misc{BYUediftools,
+title={{BYU EDIF Tools}},
+note={https://sourceforge.net/projects/byuediftools/}
+}
+
+@misc{vivado,
```
Comment (cbcunc) [inline papers/dallin_skouson/mybib.bib]: Please provide a full citation. https://www.mendeley.com/guides/web-citation-guide

**Review by cbcunc — CHANGES_REQUESTED** [review]
Minor updates to citations requested as part of final proceedings committee walk-through.

**Comment by andrewmkeller** [comment]
After subsequent discussion among the authors, a minor but appropriate title change is requested from "Performing Netlist Analysis and Transformations Using SpyDrNet" to "Netlist Analysis and Transformations Using SpyDrNet". Additional minor changes have been made to improve the clarity and flow of the paper.

File: papers/dallin_skouson/mybib.bib (hunk 1 above)
Comment (andrewmkeller) [reply papers/dallin_skouson/mybib.bib]: @cbcnc, thank your for catching this reference. A full citation has been added in 455724d.

File: papers/dallin_skouson/mybib.bib (hunk 2 above)
Comment (andrewmkeller) [reply papers/dallin_skouson/mybib.bib]: @cbcnc, thank your for catching this reference. A full citation has been added in 455724d.

File: papers/dallin_skouson/mybib.bib (hunk 3 above)
Comment (andrewmkeller) [reply papers/dallin_skouson/mybib.bib]: @cbcnc, thank your for catching this reference. A full citation has been added in 455724d.

**Review by cbcunc — APPROVED** [review]
Thank you for your updates!

**Pull request merged**